### PR TITLE
[DMS-9] demo: assertions on $Res

### PR DESCRIPTION
**Pull request by int-index**

### CODE REVIEW

**Comment by GoPavel** [comment]
O, wow, thank you

**Comment by GoPavel** [comment]
Approved.
I have pushed missing matches for `ResVarE`, but feel free to remove it if you have some objections

File: test/viper/reverse.mo [hunk 1]
```diff
@@ -2,6 +2,7 @@ actor Reverse {
   var xarray : [var Nat] = [var 1, 2, 3, 4, 5];
 
   private func copy_xarray(): [var Nat] {
+    assert:return (var:return).size() == 5;
```
Comment (GoPavel) [inline test/viper/reverse.mo:5]: One thing, I think here should be
```suggestion
    assert:return (var:return).size() == xarray.size();
```

Comment (int-index) [reply test/viper/reverse.mo:5]: Sure, makes sense

**Comment by int-index** [comment]
> Approved

Thanks for reviewing. To be clear, I don't know if we should add this `var:return` syntax, it has not been approved by Dfinity. However, I wanted to have a proof-of-concept in order to verify `reverse.mo` without errors.

**Pull request merged**